Mathlib/FieldTheory/IsSepClosed.lean

```diff
@@ -170,15 +177,72 @@ class IsSepClosure [Algebra k K] : Prop where
+  ⟨by assumption, isSeparable_self k⟩
+
 variable {k} {K}
 
 theorem isSepClosure_iff [Algebra k K] :
     IsSepClosure k K ↔ IsSepClosed K ∧ IsSeparable k K :=
   ⟨fun h => ⟨h.1, h.2⟩, fun h => ⟨h.1, h.2⟩⟩
 
-instance (priority := 100) IsSepClosure.normal [Algebra k K]
-    [IsSepClosure k K] : Normal k K :=
-  ⟨fun x => by apply IsIntegral.isAlgebraic; exact IsSepClosure.separable.isIntegral' x,
-    fun x => @IsSepClosed.splits_codomain _ _ _ _ (IsSepClosure.sep_closed k) _ _ (by
-      have : IsSeparable k K := IsSepClosure.separable
-      exact IsSeparable.separable k x)⟩
+namespace IsSepClosure
+
+instance isSeparable [Algebra k K] [IsSepClosure k K] : IsSeparable k K :=
+  IsSepClosure.separable
+
+instance (priority := 100) normal [Algebra k K] [IsSepClosure k K] : Normal k K :=
+  ⟨fun x => IsIntegral.isAlgebraic k <| IsSeparable.isIntegral' x,
+    fun x => @IsSepClosed.splits_codomain _ _ _ _ (IsSepClosure.sep_closed k) _ _
+      (have : IsSeparable k K := IsSepClosure.separable; IsSeparable.separable k x)⟩
+
+end IsSepClosure
+
+namespace IsSepClosed
+
+open IsAlgClosed lift SubfieldWithHom
+
+variable {K : Type u} {L : Type v} {M : Type w} [Field K] [Field L] [Algebra K L] [Field M]
+  [Algebra K M] [IsSepClosed M] [IsSeparable K L]
+
+theorem maximalSubfieldWithHom_eq_top : (maximalSubfieldWithHom K L M).carrier = ⊤ := by
+  rw [eq_top_iff]
+  intro x _
+  exact mem_maximalSubfieldWithHom_of_splits K L M
+    (fun x => IsIntegral.isAlgebraic K <| IsSeparable.isIntegral' x) x
+      (splits_codomain _ (IsSeparable.separable K x))
+
+/-- A (random) homomorphism from a separable extension L of K into a separably
+  closed extension M of K. -/
+noncomputable irreducible_def lift : L →ₐ[K] M := by
+  exact (maximalSubfieldWithHom K L M).emb.comp <|
+    (maximalSubfieldWithHom_eq_top (K := K) (L := L) (M := M)).symm
+      ▸ Algebra.toTop
+
+end IsSepClosed
+
+namespace IsSepClosure
+
+variable (K : Type u) [Field K] (L : Type v) (M : Type w) [Field L] [Field M]
+variable [Algebra K M] [IsSepClosure K M]
+variable [Algebra K L] [IsSepClosure K L]
+
+/-- A (random) isomorphism between two separable closures of `K`. -/
+noncomputable def equiv : L ≃ₐ[K] M :=
+  -- Porting note: added to replace local instance above
+  haveI : IsSepClosed M := IsSepClosure.sep_closed K
+  haveI : IsSeparable K L := IsSepClosure.separable
+  let f : L →ₐ[K] M := IsSepClosed.lift
+  AlgEquiv.ofBijective f
+    ⟨RingHom.injective f.toRingHom, by
+      letI : Algebra L M := RingHom.toAlgebra f
+      haveI : IsScalarTower K L M := IsScalarTower.of_algebraMap_eq <| by
+        simp only [RingHom.algebraMap_toAlgebra, RingHom.coe_coe, AlgHom.commutes, forall_const]
+      haveI : IsSepClosed L := IsSepClosure.sep_closed K
+      haveI : IsSeparable K M := IsSepClosure.separable
+      haveI : IsSeparable L M := isSeparable_tower_top_of_isSeparable K L M
+      show Function.Surjective (algebraMap L M)
+      exact IsSepClosed.algebraMap_surjective⟩
```

Comment:
I just golfed the [splitting field version of this](https://github.com/leanprover-community/mathlib4/blob/874ff0574c19582d3d275d1c803846549cd53dce//Mathlib/FieldTheory/SplittingField/Construction.lean#L362-L365); you can probably golf this the same way, as well as the [algebraic closure version](https://github.com/leanprover-community/mathlib4/blob/874ff0574c19582d3d275d1c803846549cd53dce//Mathlib/FieldTheory/IsAlgClosed/Basic.lean#L426-L440).

Reply:
Great! I golf the separable closure version. But I failed to do it for algebraic closure version because it is `L ≃ₐ[R] M` where `R` is only a commutative ring 😅 and I can't apply `algHom_bijective₂`.